main.go

```diff
@@ -471,36 +420,51 @@ func _main() error {
 		return err
 	}
 	defer os.RemoveAll(tmpDir)
-	if cacheSettings.Enabled {
-		if info, err := os.Stat(getCachedDBPath(projectID)); err == nil && info.Size() != 0 {
-			cacheSettings.CachePresent = true
+
+	if args.cacheSettings.Enabled {
+		info, err := os.Stat(getCachedDBPath(projectIdHashOrTmp))
+		args.cacheSettings.CachePresent = err == nil && info.Size() != 0
+		if !args.cacheSettings.CachePresent {
+			log.Println("Cache invalid, re-import required.")
 		}
 	}
 
 	ec := runner.NewEvalContext(*runner.DefaultSettings, tmpDir)
-	if !cacheSettings.CachePresent || !cacheSettings.Enabled {
-		err := importFiles(cacheSettings.Enabled, files, &ec, project)
-		if err != nil {
-			return err
+	// When dumping schema, need to injest even if cache is on.
+	if !args.cacheSettings.CachePresent || !args.cacheSettings.Enabled || lastNonFlagArg == "" {
```

Comment:
I think we might have to also truncate the cache file if we want to reinjest.

Reply:
Is that something I accidentally got rid of or you are just noticing it now?

Reply:
I just noticed it now. The old code didn't reingest so I didn't include that.

Reply:
The old code had to have reingested, I didn't change that.

If the cache was invalid it reingested

Reply:
Nah, it just checked that if the cache is present or not.